File: IntegrationTests/Parser/TypeParsing/FragmentUsageTests/BaseClass.cs
using InjectionCop.Fragment;

namespace InjectionCop.IntegrationTests.Parser.TypeParsing.FragmentUsageTests
{
  public class BaseClass
  {
    public virtual void Foo ([Fragment("ValidFragmentType")] int parameter1, string parameter2)
    {
    }

    public virtual void MethodWithFragmentOnSecondParameter (int parameter1, [Fragment ("ValidFragmentType")] int parameter2)
    {
    }
  }
}

File: IntegrationTests.VB/Parser/TypeParsing/FragmentUsageTests/BaseClass.vb
Imports InjectionCop.Fragment
Imports System

Namespace Parser.TypeParsing.FragmentUsageTests
	Public Class BaseClass
		Public Overridable Sub Foo(<Fragment("ValidFragmentType")> parameter1 As Integer, parameter2 As String)
		End Sub

		Public Overridable Sub MethodWithFragmentOnSecondParameter(parameter1 As Integer, <Fragment("ValidFragmentType")> parameter2 As Integer)
		End Sub
	End Class
End Namespace
